src/terminal/Process.cpp

```diff
@@ -457,8 +459,11 @@ FileSystem::path Process::homeDirectory()
         return FileSystem::path("/");
 #endif
 }
-
+#if defined(__APPLE__)
+string Process::workingDirectory(UnixPty const& _pty) const
```

Comment:
```suggestion
string Process::workingDirectory(UnixPty const* _pty = nullptr) const
```
shouldn't the parameter not made optional instead? such as above? So that the code that attempts to call this function doesn't need to have too many `#ifdef`'s?